pseudocode-defns/pseudocode_defns.tex

```diff
@@ -0,0 +1,490 @@
+\end{definition}
+
+\begin{definition}[\texttt{u32}]$\sfi$
+    u32 is the Rust 32-bit unsigned integer type. If $v$ is a value of type \texttt{u32}, then we know that $v\in \{0,1,2,\ldots, 2^{32}-1\}$. \inRust{u32}{https://doc.rust-lang.org/std/primitive.u32.html}
+\end{definition}
+
+\begin{definition}[\texttt{usize}]
+    \texttt{usize} is defined differently on 32-bit and 64-bit machines. This is because the size of this primitive is equal to the number of bytes it takes to reference any location in memory.
+    \begin{itemize}
+        \item \textbf{32-bit machines:} if \texttt{v} is a value of type \texttt{usize}, then we know that $\texttt{v}\in \{0,1,2,\ldots,2^{32} - 1\}$
+        \item \textbf{64-bit machines:} if \texttt{v} is a value of type \texttt{usize}, then we know that $\texttt{v}\in \{0,1,2,\ldots,2^{64} - 1\}$
+    \end{itemize}
+    \inRust{usize}{https://doc.rust-lang.org/std/primitive.usize.html}
+\end{definition}
+
+\begin{definition}[\texttt{Vec(T)}]$\sfi$
+    The rust type \texttt{Vec(T)} consists of ordered lists of type \texttt{T}. For example, if \texttt{T = bool}, then values of type \texttt{Vec(T)} include $\texttt{[], [0], [1], [0, 0],} \ldots$ \inRust{Vec<T>}{https://doc.rust-lang.org/std/vec/struct.Vec.html}
+\end{definition}
+
+\horizline
+
+\subsection{Notes, todos, questions}
+
+%\silvia{Discuss: bounded vs lower bound and upper bound}
+
+\todonei{Include info on MPFR, and possibly relate it to our existing definitions of floats.}
+
+\todonei{Define plus, minus, etc. below each type on which they operate. For example, the definition for \texttt{u32} should also include a definition of plus on \texttt{u32}, multiplication on \texttt{u32}, etc.}
+
+\questionr{Should we have a general definition for ``floats'' (and ``integers''?), or is it sufficiently understood what a float is in general?}
+
+\section{Domains}
+
+$\sfi$ A \emph{data domain} is a representation of the set of values on which a metric or function can operate. For example, if a function accepts inputs from the domain \texttt{IntervalDomain(1:u32,17:u32)}, this means that the function can take any input value \texttt{v} of type \texttt{u32} such that \texttt{1 <= v} and \texttt{v <= 17}.
+
+\questionr{Should anything else be included in this introduction to ``Domains''?}
+
+\begin{definition}[\texttt{AllDomain}]$\sfi$
+\texttt{AllDomain(T)} is the data domain consisting of all values of type $\T$. 
+\end{definition}
+
+For example, \texttt{AllDomain(u32)} is the domain of all values of type \texttt{u32}.
+
+\begin{definition}[\texttt{IntervalDomain}] \sfi
+For a type \texttt{T} that has a total ordering (\texttt{T} has trait \texttt{TotalOrd}) and for values \texttt{L <= U} of type \texttt{T}, \texttt{IntervalDomain(L:T, U:T)} is the domain that contains all values \texttt{v} of type \texttt{T} such that \texttt{L <= v} and \texttt{v <= U}.
```

Comment:
concept of traits hasn't been introduced yet. Move trait section before this one?

Reply:
structure issue -- we should discuss how this document should be structured. See [this reply](https://github.com/opendp/whitepapers/pull/4#discussion_r671339240) for more discussion of why alphabetization was used.

Reply:
got it, thanks

Reply:
but include cross-refs to the traits that are being referenced here

Reply:
Cross-reference added. (This is the only reference to a trait that I noticed in the "Domains" section.)